environments/openshift/observatorium.libsonnet

```diff
@@ -0,0 +1,129 @@
+                        memory: '${OBSERVATORIUM_API_MEMORY_REQUEST}',
+                      },
+                      limits: {
+                        cpu: '${OBSERVATORIUM_API_CPU_LIMIT}',
+                        memory: '${OBSERVATORIUM_API_MEMORY_LIMIT}',
+                      },
+                    },
+                  } else c
+                  for c in super.containers
+                ] + [
+                  container.new('proxy', $.observatorium.proxyImage) +
+                  container.withArgs([
+                    '-provider=openshift',
+                    '-https-address=:%d' % $.observatorium.api.service.spec.ports[1].port,
+                    '-http-address=',
+                    '-email-domain=*',
+                    '-upstream=http://localhost:%d' % $.observatorium.api.service.spec.ports[0].port,
+                    '-openshift-service-account=prometheus-telemeter',
+                    '-openshift-sar={"resource": "namespaces", "verb": "get", "name": "${NAMESPACE}", "namespace": "${NAMESPACE}"}',
+                    '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get", "name": "${NAMESPACE}", "namespace": "${NAMESPACE}"}}',
+                    '-tls-cert=/etc/tls/private/tls.crt',
+                    '-tls-key=/etc/tls/private/tls.key',
+                    '-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
+                    '-cookie-secret-file=/etc/proxy/secrets/session_secret',
+                    '-openshift-ca=/etc/pki/tls/cert.pem',
+                    '-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+                    '-skip-auth-regex=^/metrics',
+                  ]) +
+                  container.withPorts([
+                    { name: 'https', containerPort: $.observatorium.api.service.spec.ports[1].port },
+                  ]) +
+                  container.withVolumeMounts(
+                    [
+                      volumeMount.new('secret-api-tls', '/etc/tls/private'),
+                      volumeMount.new('secret-api-proxy', '/etc/proxy/secrets'),
+                    ]
+                  ),
+                ],
+              },
+            },
+          },
+        } +
+        deployment.mixin.metadata.withNamespace(namespace) +
+        deployment.mixin.spec.withReplicas('${OBSERVATORIUM_API_REPLICAS}') +
+        deployment.mixin.spec.template.spec.withServiceAccount('prometheus-telemeter') +
+        deployment.mixin.spec.template.spec.withServiceAccountName('prometheus-telemeter') +
+        deployment.mixin.spec.template.spec.withVolumes([
+          volume.fromSecret('secret-api-tls', 'api-tls'),
+          volume.fromSecret('secret-api-proxy', 'api-proxy'),
+        ]),
+    },
+  },
+} + {
+  apiVersion: 'v1',
+  kind: 'Template',
+  metadata: {
+    name: 'observatorium',
+  },
+  objects: [
+    $.observatorium.api[name]
+    for name in std.objectFields($.observatorium.api)
+  ],
+  parameters: [
```

Comment:
this part about adding the parameters is specific to the templating part; should this go in the observatorium.jsonnet, as that file is for generating the template?

Reply:
Yes, actually better idea.